packages/beacon-state-transition/src/allForks/util/epochContext.ts

```diff
@@ -373,16 +373,15 @@ export class EpochContext {
    * Return the indexed attestation corresponding to ``attestation``.
    */
   getIndexedAttestation(attestation: phase0.Attestation): phase0.IndexedAttestation {
-    const data = attestation.data;
-    const bits = Array.from(readonlyValues(attestation.aggregationBits));
-    const committee = this.getBeaconCommittee(data.slot, data.index);
-    // No need for a Set, the indices in the committee are already unique.
-    const attestingIndices: ValidatorIndex[] = [];
-    for (const [i, index] of committee.entries()) {
-      if (bits[i]) {
-        attestingIndices.push(index);
-      }
-    }
+    const {aggregationBits, data} = attestation;
+    const committeeIndices = this.getBeaconCommittee(data.slot, data.index);
+    const attestingIndices = isTreeBacked(attestation)
+      ? zipIndexesInBitList(
+          committeeIndices,
+          (attestation.aggregationBits as unknown) as TreeBacked<BitList>,
+          this.config.types.phase0.CommitteeBits
+        )
+      : committeeIndices.filter((_, index) => !!aggregationBits[index]);
```

Comment:
If this is a hot path in the state transition function we should keep using a for loop for speed

Reply:
I use `zipIndexesInBitList` to improve performance https://github.com/ChainSafe/lodestar/blob/f6778740075fe2b75edf94d1db0b5691039cb500/packages/beacon-state-transition/test/perf/util/aggregationBits.test.ts#L26

Reply:
Right! I was refering to the alternative path when it's not a tree

Reply:
If aggregationBits should always be a tree then the filter is fine